manifests/d/DeepL/DeepL/3.4.15088/DeepL.DeepL.installer.yaml

```diff
@@ -0,0 +1,23 @@
+# Created using wingetcreate 1.0.3.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.1.0.schema.json
+
+PackageIdentifier: DeepL.DeepL
+PackageVersion: 3.4.15088
+MinimumOSVersion: 10.0.0.0
+Scope: user
+InstallModes:
+- interactive
+- silent
+InstallerSwitches:
+  Silent: --silent
+  SilentWithProgress: --silent
+UpgradeBehavior: install
+Installers:
+- Architecture: x86

```

Comment:
```suggestion
- Architecture: x64
```
> One or more errors occurred. (The specified executable is not a valid application for this OS platform.)

Reply:
Interesting! x86 was set by wingetcreate AFAICT, I did not set it manually in any case. The previous manifests indeed have the x64 setting. The installed DeepL.exe seems to be 32-bit, though?

Also, Sandbox Test works fine on my end and I don't see any failed tests here. Where did those errors occur then?

Reply:
There have been a miss conception of what Architecture means, wingetcreate adds the Architecture based on the Installer, however many installers seems to be x86 but only installs x64.

As of today the Architecture needs to match the supported OS because thats how the manifests and validation pipeline works.

I'm seeing x64 all around
![VirtualBoxVM_dtcNKSTWas](https://user-images.githubusercontent.com/15158490/160996747-9cd308f6-f846-40f1-9b67-34c29a9e0867.png)
Even the Installer
![VirtualBoxVM_4QfqsDdsmc](https://user-images.githubusercontent.com/15158490/160997192-7d72785f-f0f6-4b83-aa81-c39c9315a144.png)

Reply:
I am seeing 

```
C:\Users\WDAGUtilityAccount\AppData\Local\DeepL>sigcheck.exe .\DeepL.exe

Sigcheck v2.82 - File version and signature viewer
Copyright (C) 2004-2021 Mark Russinovich
Sysinternals - www.sysinternals.com

C:\Users\WDAGUtilityAccount\AppData\Local\DeepL\DeepL.exe:
        Verified:       Signed
        Signing date:   12:56 PM 3/24/2022
        Publisher:      DeepL SE
        Company:        DeepL SE
        Description:    DeepL
        Product:        DeepL
        Prod version:   3.4.15088+fb3ba1f
        File version:   3.4.15088
        MachineType:    32-bit
``` 

and for the Installer

```
C:\Users\WDAGUtilityAccount\Downloads>sigcheck64.exe DeepLSetup.exe

Sigcheck v2.82 - File version and signature viewer
Copyright (C) 2004-2021 Mark Russinovich
Sysinternals - www.sysinternals.com

C:\Users\WDAGUtilityAccount\Downloads\DeepLSetup.exe:
        Verified:       Signed
        Signing date:   12:58 PM 3/24/2022
        Publisher:      DeepL SE
        Company:        DeepL SE
        Description:    DeepL for Windows.
        Product:        DeepL for Windows.
        Prod version:   3.4.15088
        File version:   3.4.15088
        MachineType:    32-bit
```

though the latter is irrelevant:

> wingetcreate adds the Architecture based on the Installer, however many installers seems to be x86 but only installs x64.

e.g. https://github.com/microsoft/winget-create/issues/198

> I'm seeing x64 all around

How do you see the architecture on the Task Manager screenshots? The window with the `Type` and `Install Date` columns is from Programs and Feature, I assume?

> As of today the Architecture needs to match the supported OS

So `Architecture: x64` means "runs on Windows 64-bit" (but not on 32-bit Win or any ARM), correct? But then if it is a 32-bit programme (see above), it should be able to run on both x86 and x64, should it not? Maybe I'm confused.

I think the long story short of it is: Where exactly should I double-check the architecture after I update a manifest with `wingetcreate`?

Reply:
Welp, i don't know who or what sets the MachineType, but everything points to it being wrong, might be some weirdness with the Installer Type of the Installer

> How do you see the architecture on the Task Manager screenshots? The window with the Type and Install Date columns is from Programs and Feature, I assume?

Not 100% how it works on different OS Architectures, but on x64 Windows it will say (32 bit) on any x86 software.
The Window behind is RevoUninstaller, it appears to select the Architecture based on where its Installed, e.q. `Program Files` or `Program Files (x86)` which i guess can be another indicator

> So `Architecture: x64` means "runs on Windows 64-bit" (but not on 32-bit Win or any ARM), correct? But then if it is a 32-bit programme (see above), it should be able to run on both x86 and x64, should it not? Maybe I'm confused.

Yeah the Architecture means "run on Windows X-bit", for example wingetcreate splits appx/msix into x64/x86/arm/arm64, this helps with troubleshooting

However as you know yeah, x86 can run on x64 so its valid. But then winget-cli allowed x64 installers to run on arm64, since i have no experience with arm i have no idea how well that works.

You should probably read this thread https://github.com/microsoft/winget-pkgs/pull/55052#discussion_r833910923 to see if its helps on the situation we're in 😁

> I think the long story short of it is: Where exactly should I double-check the architecture after I update a manifest with wingetcreate?

IMO if its not in the url, check Task Manager if it says (32 bit) on any of the processes.
Also if the Validation fails you can check the bot logs by going to [Checks](https://github.com/microsoft/winget-pkgs/pull/56290/checks) -> [WinGetSvc-Validation](https://dev.azure.com/ms/winget-pkgs/_build/results?buildId=298204&view=results) -> [Artifacts](https://dev.azure.com/ms/winget-pkgs/_build/results?buildId=298204&view=artifacts&pathAsName=false&type=publishedArtifacts) -> [InstallationVerificationLogs](https://dev.azure.com/ms/ed6a5dfa-6e7f-413b-842c-8305dd9e89e6/_apis/build/builds/298204/artifacts?artifactName=InstallationVerificationLogs&api-version=7.0&%24format=zip)

Reply:
Thanks!